### PR TITLE
search statistics: persist to mongo

### PR DESCRIPTION
**Pull request by yingjieMiao**
NOT FULLY TESTED YET

### CODE REVIEW

File: shoebox/app/com/keepit/common/analytics/EventListener.scala [hunk 1]
```diff
@@ -188,31 +191,42 @@ abstract class SearchUnloadListener(userRepo: UserRepo, normalizedURIRepo: Norma
...
+
+        val googleUris = (metaData \ "googleClickedURIs").asOpt[Seq[String]].getOrElse(Seq.empty[String])
+        val kifiClickedUris = (metaData \ "kifiClickedURIs").asOpt[Seq[String]].getOrElse(Seq.empty[String])
+        val kifiShownUris = (metaData \ "kifiShownURIs").asOpt[Seq[String]].getOrElse(Seq.empty[String])
+        val uuid = (metaData \ "queryUUID").asOpt[String].get
+        val queryString = (metaData \ "query").asOpt[String].get
+
+        val (userId, kifiClickedIds, kifiShownIds, googleClickedIds) = db.readOnly { implicit s =>
+          val userId = userRepo.get(extUserId).id.get
+          val kifiClickedIds = kifiClickedUris.map { normalizedURIRepo.getByNormalizedUrl(_) }.filter(_ != None).map(_.get.id).filter(_ != None).map(_.get)
```
Comment (ymatsuda) [inline shoebox/app/com/keepit/common/analytics/EventListener.scala]: kifiClickedUris.flatMap{ normalizedURIRepo.getByNormalizedUrl(_) }.flatMap(_.id).map(_.get)

File: shoebox/app/com/keepit/common/analytics/EventListener.scala [hunk 2]
```diff
@@ -26,10 +26,15 @@ import akka.actor.Props
...
+import com.keepit.search.TrainingDataLabeler
+import com.keepit.search.SearchStatisticsExtractorFactory
+import com.keepit.search.UriLabel
+import com.keepit.serializer.SearchStatisticsSerializer
 
 abstract class EventListenerPlugin(
-    userRepo: UserRepo,
-    normalizedURIRepo: NormalizedURIRepo)
+  userRepo: UserRepo,
+  normalizedURIRepo: NormalizedURIRepo)
```
Comment (yingjieMiao) [inline shoebox/app/com/keepit/common/analytics/EventListener.scala:37]: sorry for the space changes from auto-formatter.

File: shoebox/app/com/keepit/controllers/search/SearchStatisticsController.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  sseFactory: Provider[SearchStatisticsExtractorFactory],
+  persistEventProvider: Provider[PersistEventPlugin],
+  implicit private val clock: Clock,
+  implicit private val fortyTwoServices: FortyTwoServices)
+  extends SearchServiceController {
+
+  def persistSearchStatistics() = Action(parse.json) { request =>
+    val json = request.body
+
+    val queryUUID = (json \ "queryUUID").asOpt[String].get
```
Comment (gmethvin) [inline shoebox/app/com/keepit/controllers/search/SearchStatisticsController.scala]: `asOpt[String].get` is basically the same thing as `as[String]`

File: shoebox/app/com/keepit/search/SearchStatisticsExtractor.scala [hunk 4]
```diff
@@ -289,7 +299,7 @@ class SearchStatisticsHelperSearcher (queryString: String, userId: Id[User], tar
   def getUriInfo() = {
 
     var uriInfoMap = Map.empty[Id[NormalizedURI], UriInfo]
-    val idsetFilter = new IdSetFilter(targetUriIds.map{ normalizedId => normalizedId.id})
+    val idsetFilter = new IdSetFilter(targetUriIds.map{ normalizedId => normalizedId.id}.toSet)
```
Comment (gmethvin) [inline shoebox/app/com/keepit/search/SearchStatisticsExtractor.scala]: `_.id`

**Pull request merged**